nautobot/core/templates/inc/nav_menu.html

```diff
@@ -837,6 +847,34 @@
                     </span>
                 </div>
             </form>
+            {% comment %}
+            The following if statement means the Query tab is only rendered when the GraphiQL
+            is rendered.
+            {% endcomment %}
+            {% if  graphiql and perms.extras.view_graphqlquery %}
+
+                <ul class="nav navbar-nav navbar-right">
+                    <li class="dropdown">
+                        <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">
+                            <span>Queries </span>
+                            <span class="caret"></span>
+                        </a>
+                        <ul class="dropdown-menu" id="graphql-queries-menu">
+                            {% for query in saved_graphiql_queries %}
+                                <li>
+                                    <a href="#" onclick="reload('{{query.id}}', '{{ query.query|urlencode }}', '{{ query.variables|render_json|urlencode }}')">{{ query.name }}</a>
+                                </li>
+                            {% endfor %}
+                            <br>
+                            <div style="margin:0 auto; display: block; width: fit-content; height: fit-content">
+                                <button type="button" class="btn btn-warning" onclick="save()"><i class="mdi mdi-content-save-edit"></i> Save</button>
+                                <button type="button" class="btn btn-success" onclick="populate_model()" data-toggle="modal" data-target="#GraphQLQueries_form" title="Query Form"><i class="mdi mdi-content-save"></i> Save As</button>
+                            </div>
+                        </ul>
+                    <li>
+                </ul>
```

Comment:
Here's a possible suggestion for an alternate layout of this menu:

![image](https://user-images.githubusercontent.com/5603551/119662565-87590380-bdff-11eb-8451-42a7ac46a930.png)

The idea being that the currently selected query is `<strong>` and the `Save Changes` button appears on the same line as that query to make it explicit what will be saved. Note also the addition of the gray text and divider line to more closely match other Nautobot nav menus.

HTML:

![image](https://user-images.githubusercontent.com/5603551/119662829-d2731680-bdff-11eb-99b0-474ad38ac64a.png)